File: sert-2014/src/edu/wpi/first/wpilibj/templates/commands/ChangeArmPosition.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package edu.wpi.first.wpilibj.templates.commands;

import edu.wpi.first.wpilibj.Timer;

/**
 *
 * @author SERT
 */
public class ChangeArmPosition extends CommandBase {
    

    public ChangeArmPosition() {
        // Use requires() here to declare subsystem dependencies
        // eg. requires(chassis);
        requires(armSub);
        
    }

    // Called just before this Command runs the first time
    protected void initialize() {
        
    }

    // Called repeatedly when this Command is scheduled to run
    protected void execute() {
    }

    // Make this return true when this Command no longer needs to run execute()
    protected boolean isFinished() {
        return true;
    }

    // Called once after isFinished returns true
    protected void end() {
        if (armSub.isUp()) {
            armSub.lowerArm();
        } else {
            armSub.raiseArm();          
        }
        Timer.delay(.01);               //minimum time for Solenoid to switch positions
        armSub.resetArmSolenoid();      //sets solenoid to off to prevent burning it out
    }
    
    protected void interrupted() {
    }
}

File: sert-2014-C++/Commands/ChangeArmPosition.cpp
#include "ChangeArmPosition.h"

ChangeArmPosition::ChangeArmPosition()
{
	Requires( m_armSub );
}

// Called just before this Command runs the first time
void ChangeArmPosition::Initialize()
{
}

// Called repeatedly when this Command is scheduled to run
void ChangeArmPosition::Execute() 
{
}

// Make this return true when this Command no longer needs to run execute()
bool ChangeArmPosition::IsFinished() 
{
	return true;
}

// Called once after isFinished returns true
void ChangeArmPosition::End() 
{
    if ( m_armSub->IsUp() ) 
    {
    	m_armSub->LowerArm();
    } 
    else 
    {
    	m_armSub->RaiseArm();          
    }
}

// Called when another command which requires one or more of the same
// subsystems is scheduled to run
void ChangeArmPosition::Interrupted() 
{
}

